components/builder-graph/src/data_store.rs

```diff
@@ -69,25 +274,156 @@ impl DataStore {
         }
 
         for package in rows {
-            packages.push(package);
+            packages.push(PackageInfo::from(package));
         }
 
         Ok(packages)
     }
 
-    pub fn get_job_graph_package(&self,
-                                 ident: &PackageIdent,
-                                 target: PackageTarget)
-                                 -> Result<PackageWithVersionArray> {
+    fn get_job_graph_package(&self,
+                             ident: &PackageIdent,
+                             target: PackageTarget)
+                             -> Result<PackageWithVersionArray> {
         let conn = self.pool.get_conn()?;
 
         let package = GetLatestPackage { ident:      BuilderPackageIdent(ident.clone()),
                                          target:     BuilderPackageTarget(target),
                                          visibility: PackageVisibility::all(), };
 
-        println!("Package fetching: {:?}", package);
+        debug!("get_job_graph_package fetching: {} {}", ident, target);
 
         let package = Package::get_latest(package, &conn).map_err(Error::DieselError)?;
         Ok(package)
     }
+
+    fn get_origin_channel_latest(&self,
+                                 origin: &str,
+                                 channel: &str,
+                                 target: PackageTarget)
+                                 -> Result<Vec<PackageIdent>> {
+        let conn = self.pool.get_conn()?;
+
+        let query = origin_packages_with_version_array::table
+            .inner_join(origin_channel_packages::table.inner_join(origin_channels::table))
+            .select(origin_packages_with_version_array::ident)
+            .distinct_on((
+                origin_packages_with_version_array::name,
+                origin_packages_with_version_array::target,
+            ))
+            .order(sql::<PackageWithVersionArray>(
+                "origin_packages_with_version_array.name, \
+        origin_packages_with_version_array.target, \
+        string_to_array(origin_packages_with_version_array.version_array[1],'.')::\
+        numeric[] desc, origin_packages_with_version_array.ident_array[4] desc",
+            ))
+            .filter(origin_channels::name.eq(&channel))
+            .filter(origin_packages_with_version_array::origin.eq(&origin))
+            .filter(origin_packages_with_version_array::target.eq(target.to_string()));
+
+        let result: Vec<BuilderPackageIdent> = query.get_results(&conn).unwrap();
```

Comment:
As above is this unwrap safe? As a follow up, I checked on some of our other database query functions to verify the behavior, it seems like we typically hand off the result of the diesel query to the calling function inside of a match. Is the unwrap necessary? https://github.com/habitat-sh/builder/blob/049a18bd2d6694fc4ea9344805e9010d1f9edb79/components/builder-db/src/models/package.rs#L447